code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -244,6 +244,20 @@
 		"<span class='userdanger'>You feel a powerful shock coursing through your body!</span>", \
 		"<span class='italics'>You hear a heavy electrical crack.</span>" \
 		)
+	if(iscarbon(pulling) && !illusion && source != pulling)
+		if(ishuman(pulling))
+			var/mob/living/carbon/human/H = pulling
+			H.electrocute_act(shock_damage*0.75, src, 1, safety, override, tesla_shock, illusion, stun)
+		else
+			var/mob/living/carbon/C = pulling
+			C.electrocute_act(shock_damage*0.75, src, 1, safety, override, tesla_shock, illusion, stun)

```

Comment:
am I missing something? why is human/carbon separate checks but do the same?

Reply:
Humans have their own electrocute_act, which calls the carbon_defense electrocute act at the end. It's just me faffing about with casting, I never did quite manage to fully wrap my head around casting. It seems to work fine when casting a carbon to human, so I'll remove the extra lines.